[Docs] Ensure breaking-changes tag exists for empty and existing breaking changes

craigtaverner opened the pull request:
The original feature only added the tags when there were breaking changes with `notable: true`. Last year we improved this by adding the tag when there were no breaking changes (in 8.3.0). However we missed the case where there are breaking changes, but the `notable: false` means the tags did not get added. This fixes that case.

I manually tested this with three cases:

* No breaking changes
* Only one notable=false
* One notable:false and one notable:true

Fixes #94910

craigtaverner committed:
Ensure breaking-changes tag exists for both cases

This was previously added so that it only worked when there were no
breaking changes, but should be for the case of breaking changes also

abdonpijpelink commented on build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc:
```diff
@@ -15,10 +15,10 @@ coming::[${majorDotMinorDotRevision}]
 === Breaking changes
-<% if (breakingByNotabilityByArea.isEmpty()) { %>
+
 // tag::notable-breaking-changes[]
```
If there are breaking changes, the opening `// tag::notable-breaking-changes[]` tag should be lower, on line 27 (after the "The following changes.." paragraph. See for example: https://github.com/elastic/elasticsearch/blob/main/docs/reference/migration/migrate_8_5.asciidoc?plain=1#L22

craigtaverner replied:
Looking further down I see there is already a notable-breaking-changes tag. So this fix is not valid. The alternative explanation is that there are no breaking changes with the `notable` flag. We need an alternative fix.

abdonpijpelink reviewed (commented):
I've left a comment

craigtaverner committed:
Revert "Ensure breaking-changes tag exists for both cases"

This reverts commit f8cb87ad45eeeb5458ceba20fdee30d82ae62a05.

craigtaverner committed:
Ensure breaking-changes tag exists for both cases

This was previously added so that it only worked when there were no
breaking changes, but should be for the case of breaking changes also

craigtaverner committed:
Improved wording

abdonpijpelink reviewed (approved):
LGTM. Thanks @craigtaverner !

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.7](https://github.com/elastic/elasticsearch/pull/94932)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/94932">](https://github.com/elastic/elasticsearch/pull/94932) |
